File: 이코테 복습(python, c++, java)/구현/c++/문자열 재정렬.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

string str;
vector<char>result;
int num = 0;

int main() {
	cin >> str;
	for (int i = 0; i < str.size(); i++) {
		if (isalpha(str[i])) {
			result.push_back(str[i]);
		}
		else {
			num += str[i] - '0';
		}
	}
	sort(result.begin(), result.end());
	for (int i = 0; i < result.size(); i++) {
		cout << result[i];
	}
	if (num != 0) {
		cout << num;
	}
	cout << endl;

}

File: 이코테 복습(python, c++, java)/구현/python/문자열 재정렬.py
s = input()
alpha = []
num = 0
for i in range(len(s)):
    if s[i].isalpha():
        alpha.append(s[i])
    else:
        num += int(s[i])
alpha.sort()
if num != 0:
    alpha.append(str(num))
answer = ''.join(alpha)
print(answer)
